lib/cli.js

```diff
@@ -98,7 +99,15 @@ if (options.debug) {
     debug('cli.input', cli.input);
 }
 
-npmCheck(options)
+(
+    process.env.NPM_CHECK_INSTALLER ?
+        Promise.resolve(process.env.NPM_CHECK_INSTALLER) :
+        detectPreferredPM(options.cwd)
+)
+    .then(preferredPM => {
+        options.installer = preferredPM && preferredPM.name === 'pnpm' ? 'pnpm' : 'npm';
```

Comment:
You are potentially checking `.name` of a string here 🤔

Reply:
Actually, not a super fan of this...

Could we do it like this:

1. Change the default `installer` from `npm` to `auto` (on line 90/91)

  `installer: process.env.NPM_CHECK_INSTALLER || 'auto',`

2. Change the resolving to something like this:

  ```js
  Promise.resolve()
    .then(() => {
      return (options.installer === 'auto' ? detectPreferredPM(options.cwd) : options.installer)
    })
    .then((installer) => {
      return npmCheck(Object.assign(options, { installer }))
    })